cloudflare_exporter/collector.py

```diff
@@ -167,10 +171,30 @@ def __init__(self):
         self.received_requests_pop_origin = Counter()
         self.origin_response_times = defaultdict(list)
 
+    @staticmethod
+    @lru_cache(maxsize=1000)
```

Comment:
Why is this cache needed? (especially if cleaned at each run)

Reply:
because we call this function on each log line, and I don't want to make a dns call for each line !